### PR TITLE
Add Cosmos database resource.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR adds support for `AddDatabase(...)` on Cosmos database resources.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1583)

Usage:

```csharp
var builder = DistributedApplication.CreateBuilder(args)
var account = builder.AddCosmosDB("myaccount");
var db = account.AddDatabase("imports");
```

The manifest will look like this:

```json
{
  "resources": {
    "myaccount": {
      "type": "azure.cosmosdb.account.v0"
    },
    "imports": {
      "type": "azure.cosmosdb.database.v0",
      "parent": "myaccount"
    }
  }
}
```

The expectation is that a deployment tool that supports these resource types would create the database as well.

### CODE REVIEW

**Comment by davidfowl** [comment]
Can you update the provisioner as well?

File: src/Aspire.Hosting.Azure/AzureCosmosDBCloudApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -29,6 +29,19 @@ public static class AzureCosmosDBCloudApplicationBuilderExtensions
                       .WithManifestPublishingCallback(context => WriteCosmosDBToManifest(context, connection));
     }
 
+    public static IResourceBuilder<AzureCosmosDBDatabaseResource> AddDatabase(this IResourceBuilder<AzureCosmosDBResource> builder, string name)
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure/AzureCosmosDBCloudApplicationBuilderExtensions.cs:32]: XML docs?

**Review by Pilchie — APPROVED** [review]
This is essentially exactly what I had in one of the iterations of my PR ;)

**Comment by mitchdenny** [comment]
Yeah at the time I didn't quite realize just how painful it is to create a database when using managed identity with the data plane SDK.

**Comment by Pilchie** [comment]
Note that I think in the long term we're going to loosen that up and support it in the data plane SDK, but that's probably a next semester work item.

**Comment by mitchdenny** [comment]
> Can you update the provisioner as well?

Provisioner updated.

File: src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureCosmosDBProvisioner.cs [hunk 2]
```diff
@@ -76,5 +93,36 @@ public override bool ConfigureResource(IConfiguration configuration, AzureCosmos
 
         var connectionStrings = context.UserSecrets.Prop("ConnectionStrings");
         connectionStrings[resource.Name] = resource.ConnectionString;
+
+        foreach (var database in resource.Databases)
+        {
+            await CreateDatabaseIfNotExists(cosmosResource, database, cancellationToken).ConfigureAwait(false);
+            connectionStrings[database.Name] = resource.ConnectionString; // Same as parent resource.
+        }
+    }
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureCosmosDBProvisioner.cs:102]: You're not handling deleting unused ones. Take a look at what service bus does for queues and topics.

Comment (mitchdenny) [reply src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureCosmosDBProvisioner.cs:113]: Done.

File: src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureCosmosDBProvisioner.cs [hunk 3]
```diff
@@ -94,6 +94,17 @@ public override bool ConfigureResource(IConfiguration configuration, AzureCosmos
         var connectionStrings = context.UserSecrets.Prop("ConnectionStrings");
         connectionStrings[resource.Name] = resource.ConnectionString;
 
+        var existingDatabases = cosmosResource.GetCosmosDBSqlDatabases().GetAllAsync(cancellationToken);
+
+        await foreach (var existingDatabase in existingDatabases)
+        {
+            if (!resource.Databases.Any(d => d.Name == existingDatabase.Data.Name))
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureCosmosDBProvisioner.cs:101]: You could do a single pass over the set of databases but I'm not concerned about performance 😄

**Review by davidfowl — APPROVED** [review]

**Pull request merged**